mlflow/projects/_project_spec.py

```diff
@@ -0,0 +1,136 @@
+"""Internal utilities for parsing MLproject YAML files."""
+
+import os
+
+from six.moves import shlex_quote
+
+from mlflow import data
+
+class Project(object):
+    """A project specification loaded from an MLproject file."""
+    def __init__(self, uri, yaml_obj):
+        self.uri = uri
+        self.name = os.path.splitext(os.path.basename(os.path.abspath(uri)))[0]
+        self.conda_env = yaml_obj.get("conda_env")
+        self.entry_points = {}
+        for name, entry_point_yaml in yaml_obj.get("entry_points", {}).items():
+            parameters = entry_point_yaml.get("parameters", {})
+            command = entry_point_yaml.get("command")
+            self.entry_points[name] = EntryPoint(name, parameters, command)
+        # TODO: validate the spec, e.g. make sure paths in it are fine
+
+    def get_entry_point(self, entry_point):
+        from mlflow.projects import ExecutionException
```

Comment:
This change was to avoid recursive imports?

Reply:
Yes, that's why.